sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/CatalystTypeConverters.scala

```diff
@@ -158,6 +158,20 @@ object CatalystTypeConverters {
 
     override def toCatalystImpl(scalaValue: Any): ArrayData = {
       scalaValue match {
+        case a: Array[Boolean] =>
+          UnsafeArrayData.fromPrimitiveArray(a)
+        case a: Array[Byte] =>
+          UnsafeArrayData.fromPrimitiveArray(a)
+        case a: Array[Short] =>
+          UnsafeArrayData.fromPrimitiveArray(a)
+        case a: Array[Int] =>
+          UnsafeArrayData.fromPrimitiveArray(a)
+        case a: Array[Long] =>
+          UnsafeArrayData.fromPrimitiveArray(a)
+        case a: Array[Float] =>
+          UnsafeArrayData.fromPrimitiveArray(a)
+        case a: Array[Double] =>
+          UnsafeArrayData.fromPrimitiveArray(a)
```

Comment:
yea let's remove them, `CatalystTypeConverter` will not be in critical path and we don't need to optimize it.